### PR TITLE
[STAL-2472] feat: avoid processing minified JavaScript files

### PR DESCRIPTION
**Pull request by amaanq**
## What problem are you trying to solve?

Currently, we process minified JavaScript files during analysis. These files are very expensive to parse and query, and can raise false positives. Ideally, we should not be processing them to begin with, as these are not hand-written files by programmers.

## What is your solution?

When we iterate over each language to process and fetch the files to be analyzed for the language, an additional step is added where we filter out minified files. The filtering process is done in the function `filter_out_minified_files`, and the heuristic for determining whether or not a file is minified is by checking if the average line length of the file is over 110. This is currently what GitHub does with `linguist` [here](https://github.com/github-linguist/linguist/blob/ebbedf06dc18e929104a4a3573ac43b43298a761/lib/linguist/generated.rb#L168). A test has been added where we write to a JavaScript file a few hundred characters on one line, and after filtering out minified files, we can see that the file was filtered out.

### CODE REVIEW

File: crates/bins/src/bin/datadog-static-analyzer.rs [hunk 1]
```diff
@@ -471,7 +471,11 @@ fn main() -> Result<()> {
     let mut number_of_rules_used = 0;
     // Finally run the analysis
     for language in &languages {
-        let files_for_language = filter_files_for_language(&files_to_analyze, language);
+        let mut files_for_language = filter_files_for_language(&files_to_analyze, language);
+        if language == &Language::JavaScript {
+            files_for_language =
+                filter_out_minified_files(&files_for_language, directory_path, use_debug);
+        }
```
Comment (juli1) [inline crates/bins/src/bin/datadog-static-analyzer.rs:478]: 1. Implement this rather as the `is_generated_file`. Look at [this](https://github.com/DataDog/datadog-static-analyzer/blob/main/crates/static-analysis-kernel/src/analysis/analyze.rs#L186) and we should implement something similar.
2. This should be enabled only when `analysis_option.ignore_generated_files` is true (similar to (1))

**Review by juli1 — CHANGES_REQUESTED** [review]

**Comment by amaanq** [comment]
The CI failures are expected - we're no longer processing minified JS files in repos that we use to test for regressions in.

File: crates/static-analysis-kernel/src/analysis/generated_content.rs [hunk 2]
```diff
@@ -49,6 +49,22 @@ pub fn is_generated_file(full_content: &str, language: &Language) -> bool {
     }
 }
 
+/// Returns if a file is minified or not.
+/// The heuristic for detecting minified files is based on the average line length being greater
+/// than 110.
+pub fn is_minified_file(content: &str, language: &Language) -> bool {
+    if language == &Language::JavaScript {
+        let lines = content.lines();
+        if lines.count() == 0 {
```
[the remark this answers is not in the document]
Comment (amaanq) [reply crates/static-analysis-kernel/src/analysis/generated_content.rs:58]: Your example with `checked_div` is much more clear & easier to read, so I'll use that, thanks!

**Review by juli1 — APPROVED** [review]

**Comment by juli1** [comment]
Please fix the CI pipeline + ship

**Pull request merged**